renderlib/gesture/gesture.cpp

```diff
@@ -628,8 +629,15 @@ Gesture::drawCircle(glm::vec3 center,
     glm::vec3 p0 = center + xaxis * cosf(theta0) + yaxis * sinf(theta0);
     glm::vec3 p1 = center + xaxis * cosf(theta1) + yaxis * sinf(theta1);
 
-    graphics.addLine(Gesture::Graphics::VertsCode(p0, color, opacity, code),
-                     Gesture::Graphics::VertsCode(p1, color, opacity, code));
+    if (clipPlane) {
+      if (glm::dot(*clipPlane, glm::vec4(p0, 1.0)) > 0 && glm::dot(*clipPlane, glm::vec4(p1, 1.0)) > 0) {
+        graphics.addLine(Gesture::Graphics::VertsCode(p0, color, opacity, code),
+                         Gesture::Graphics::VertsCode(p1, color, opacity, code));
+      }
```

Comment:
Yeah this is admittedly crude but it works if the circles use small enough segments.   I'll refine it as needed for future usage.  Currently there's nothing else that needs to clip, yet.

Reply:
It's also possible to use gpu clip planes and have this all done in the shader with good interpolation, but it would result in potentially splitting up the geometry into more draw calls. Currently the code can accumulate lots of line segments into a single draw call even if they are drawing many different things.   And while things are still super cheap to draw right now, I didn't feel like routing all the data into the shader and splitting draw calls if they need different clip planes.